### ISSUE TITLE
Feature Request: Element-wise transcendental functions (log2, ln, exp) for SIMD entropy calculation

### ISSUE THREAD

**Issue by noahgift** [paiml/trueno#1]
## Summary

Add element-wise transcendental functions to `Vector<f32>` to enable SIMD-accelerated entropy and information-theoretic calculations.

## Motivation

While integrating Trueno into PMAT's similarity detection module, I implemented SIMD-accelerated cosine similarity using `Vector::dot()` and `Vector::norm_l2()` - these work excellently.

However, **Shannon entropy calculation** requires element-wise `log2`:

```rust
// Current scalar implementation (can't use Trueno)
fn entropy_scalar(probabilities: &[f64]) -> f64 {
    let mut entropy = 0.0;
    for &p in probabilities {
        if p > 0.0 {
            entropy -= p * p.log2();  // Need vectorized log2
        }
    }
    entropy
}
```

Without element-wise `log2`, entropy calculation must remain scalar, losing potential 2-8x SIMD speedup.

## Proposed API

```rust
impl Vector<f32> {
    /// Element-wise natural logarithm
    pub fn ln(&self) -> Result<Self>;
    
    /// Element-wise base-2 logarithm
    pub fn log2(&self) -> Result<Self>;
    
    /// Element-wise base-10 logarithm  
    pub fn log10(&self) -> Result<Self>;
    
    /// Element-wise exponential (e^x)
    pub fn exp(&self) -> Result<Self>;
    
    /// Element-wise power
    pub fn pow(&self, exponent: f32) -> Result<Self>;
}
```

## Use Cases

1. **Shannon Entropy**: `-sum(p * log2(p))` - information theory
2. **Cross-Entropy Loss**: ML training loss functions
3. **Softmax**: `exp(x) / sum(exp(x))` - neural network activation
4. **Log-likelihood**: Statistical models
5. **KL Divergence**: Distribution comparison

### SIMD Considerations

Transcendental functions are challenging for SIMD because they typically require:
- Polynomial approximations (minimax or Chebyshev)
- Range reduction
- Special case handling (NaN, Inf, negative inputs for log)

### Recommended Approach

1. **Use `sleef-rs`** or similar for tested SIMD transcendental implementations
2. **Or** implement using polynomial approximations:
   - `log2(x) ≈ exponent + polynomial(mantissa)` using IEEE754 bit manipulation
   - Accuracy: ~1-2 ULP for most inputs

### Reference Implementation

Intel's SVML and AMD's LibM provide optimized SIMD transcendentals. For Rust:
- `packed_simd` has some transcendentals
- `sleef-rs` wraps SLEEF library
- `fast-math` crate has approximations

## Priority

**Medium-High** - Enables a significant class of SIMD optimizations (information theory, ML)

## Related Work

- PMAT integration: `server/src/services/similarity.rs` - entropy calculation currently scalar
- Trueno spec: `docs/specifications/trueno-integration-spec.md`

## Acceptance Criteria

- [ ] `Vector::log2()` returns element-wise base-2 logarithm
- [ ] `Vector::ln()` returns element-wise natural logarithm  
- [ ] `Vector::exp()` returns element-wise exponential
- [ ] Handles edge cases: log(0) → -Inf, log(negative) → NaN
- [ ] SIMD acceleration on x86 (AVX2) and ARM (NEON)
- [ ] Property tests for accuracy (within 2 ULP of scalar)
- [ ] Benchmarks showing speedup vs scalar

## Labels

`enhancement`, `simd`, `transcendentals`
